lnrpc/walletrpc/walletkit.proto

```diff
@@ -300,5 +308,53 @@ service WalletKit {
     fee preference being provided. For now, the responsibility of ensuring that
     the new fee preference is sufficient is delegated to the user.
     */
-    rpc BumpFee(BumpFeeRequest) returns (BumpFeeResponse);
+    rpc BumpFee (BumpFeeRequest) returns (BumpFeeResponse);
+
+    /*
+    DeriveSharedKey returns a shared secret key by performing Diffie-Hellman key
+    derivation between the ephemeral public key in the request and the node's
+    identity public key. The resulting shared public key is hashed with sha256
+    and the final key length therefore is 256bit.
+    
+    If any sender wants to encrypt a message that only the receiving lnd node
+    can decrypt, the Diffie-Hellman key exchange is very useful. As the
+    recipient's public key is already knows as the nodes' identity public key in
+    the network, no two-way key exchange is necessary.
+    All the sender has to do is the following:
+       1. Create a new temporary/ephemeral private key d_send and its
+          corresponding public key P_send.
+
+          P_send = d_send * G
+
+       2. Take the recipient's identity public key P_recipient and scalar
+          multiply it with the ephemeral private key d_send to arrive at the
+          shared public key P_shared.
+
+          P_shared = d_send * P_recipient
+
+       3. Serialize the shared public key P_shared in the compressed format and
+          run it through the sha256 hash function to arrive at the 256bit shared
+          secret key d_sec that can be used for any symmetrical encryption
+          algorithm like AES.
+       4. Encrypt the message m with the shared secret key d_sec and send it to
+          the receiver together with the ephemeral public key P_send.
+
+    The recipient can arrive at the same shared secret key d_sec by calling this
+    RPC function which does the following:
+       1. Take the ephemeral public key P_send from the request and multiply it
+          with the nodes' identity private key d_recipient. Thanks to the
+          associativity of elliptic curves, this is the same as multiplying
+          the sender's private key with the recipient's public key.
+
+          P_shared = d_recipient * P_send
+          P_shared = d_recipient * (d_send * G)
+          P_shared = d_send * (d_recipient * G)
+          P_shared = d_send * P_recipient
+
+      2. Serialize the shared public key P_shared in the compressed format and
+         sha256 hash it to arrive at the shared secret key.
+      3. Use the derived key to decrypt the message using a symmetrical
+         decryption algorithm.
+    */
```

Comment:
I don't know if we should include this whole explanation of DH. I think it is, similar to `SignMessage` upon the caller to understand what they should do.